Sources/MongoSwiftSync/MongoClient.swift

```diff
@@ -34,7 +41,28 @@ public class MongoClient {
      *     built with TLS support.
      */
     public init(_ connectionString: String = "mongodb://localhost:27017", options: ClientOptions? = nil) throws {
-        fatalError("unimplemented")
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 5)
+        do {
+            self.asyncClient = try MongoSwift.MongoClient(connectionString, using: eventLoopGroup, options: options)
+            self.eventLoopGroup = eventLoopGroup
+        } catch {
+            try eventLoopGroup.syncShutdownGracefully()
+            throw error
+        }
+    }
+
+    deinit {
+        do {
```

Comment:
I'm not sure if there is really a better way to handle these.

I wrote it this way in separate try/catches so that even if closing the client fails we will still shut down the event loop group.

Reply:
Once we have swift-log integration, I think this would be a good place to do some error logging. Besides that, there's not really much else we can do from deinit in terms of error reporting, so until then this lgtm.